spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/factory/HystrixGatewayFilterFactory.java

```diff
@@ -100,11 +102,36 @@ public GatewayFilter apply(Config config) {
 				s.onCancel(sub::unsubscribe);
 			}).onErrorResume((Function<Throwable, Mono<Void>>) throwable -> {
 				if (throwable instanceof HystrixRuntimeException) {
+					/*
+					 * Special handling of error events that need to be mapped to specific status codes,
+					 * instead of a generic 500
+					 */
 					HystrixRuntimeException e = (HystrixRuntimeException) throwable;
-					if (e.getFailureType() == TIMEOUT) {
-						return Mono.error(new ResponseStatusException(HttpStatus.GATEWAY_TIMEOUT));
+
+					switch (e.getFailureType()) {
+
+						case TIMEOUT:
+							return Mono.error(new ResponseStatusException(HttpStatus.GATEWAY_TIMEOUT));
+
+						case COMMAND_EXCEPTION:
+							Throwable cause = e.getCause();
+
+							if (cause instanceof ResponseStatusException) {
+								/*
+								 * Since ResponseStatusException already carries the desired status
+								 * code to be transmitted to the client, just propagate it.
+								 */
+								return Mono.error(cause);
+							}
+
+							if (cause instanceof TimeoutException) {
```

Comment:
When spring-projects/spring-boot#14742 is fixed upstream, we can just add `@ResponseStatus` to `TimeoutException` and let the previous case (amended as per comment) take care of propagating the cause outside.